src/Microsoft.DocAsCode.Build.Engine/SingleDocumentBuilder.cs

```diff
@@ -580,8 +580,12 @@ private void FeedXRefMap(List<ManifestItemWithContext> manifest, IDocumentBuildC
                 using (new LoggerFileScope(m.FileModel.LocalPathFromRepoRoot))
                 {
                     Logger.LogDiagnostic($"Feed xref map from template for {m.Item.DocumentType}...");
-                    var bookmarks = m.Options.Bookmarks;
-                    // TODO: Add bookmarks to xref
+                    // TODO: use m.Options.Bookmarks directly after all templates report bookmarks
+                    var bookmarks = m.Options.Bookmarks ?? m.FileModel.Bookmarks;
```

Comment:
>m.Options.Bookmarks [](start = 36, length = 19)

Question: the method FeedXRefMap seems to run before templateProcessor.Process? then how could templates collect bookmark info?